test/fin_fold.lean

```diff
@@ -0,0 +1,5 @@
+import Std.Data.Fin.Basic
+import Std.Tactic.GuardMsgs
```

Comment:
```suggestion
import Std.Tactic.GuardExpr
```
suffices.

Not sure how I feel about having a test file for these, when we really want a lemma!